rotkehlchen/chain/ethereum/modules/curve/decoder.py

```diff
@@ -228,64 +235,190 @@ def _decode_curve_events(  # pylint: disable=no-self-use
         all_logs: list[EvmTxReceiptLog],  # pylint: disable=unused-argument
         action_items: Optional[list[ActionItem]],  # pylint: disable=unused-argument
     ) -> tuple[Optional[HistoryBaseEntry], list[ActionItem]]:
-        if tx_log.topics[0] in (
-            REMOVE_LIQUIDITY,
-            REMOVE_ONE,
-            REMOVE_LIQUIDITY_IMBALANCE,
-            REMOVE_LIQUIDITY_3_ASSETS,
-            REMOVE_LIQUIDITY_4_ASSETS,
-        ):
-            user_address = hex_or_bytes_to_address(tx_log.topics[1])
+        if tx_log.topics[0] in REMOVE_LIQUIDITY_EVENTS:
+            maybe_user_address = hex_or_bytes_to_address(tx_log.topics[1])
             return self._decode_curve_remove_events(
                 tx_log=tx_log,
                 transaction=transaction,
                 decoded_events=decoded_events,
-                user_address=user_address,
+                maybe_user_address=maybe_user_address,
             )
-        if tx_log.topics[0] in (
-            ADD_LIQUIDITY,
-            ADD_LIQUIDITY_2_ASSETS,
-            ADD_LIQUIDITY_4_ASSETS,
-        ):
-            user_address = hex_or_bytes_to_address(tx_log.topics[1])
+        if tx_log.topics[0] in ADD_LIQUIDITY_EVENTS:
+            maybe_user_address = hex_or_bytes_to_address(tx_log.topics[1])
             return self._decode_curve_deposit_events(
                 transaction=transaction,
                 tx_log=tx_log,
                 decoded_events=decoded_events,
-                user_address=user_address,
+                maybe_user_address=maybe_user_address,
             )
 
         return None, []
 
-    @staticmethod
-    def _maybe_enrich_curve_transfers(
-            token: EvmToken,  # pylint: disable=unused-argument
-            tx_log: EvmTxReceiptLog,
+    def _handle_post_decoding(
+            self,
+            transaction: EvmTransaction,
+            decoded_events: list[HistoryBaseEntry],
+            all_logs: list[EvmTxReceiptLog],  # pylint: disable=unused-argument
+    ) -> list[HistoryBaseEntry]:
+        """This method handles post decoding for curve events made through the deposit contracts.
+
+        This is required because certain events are not available when `AddLiquidity` or `RemoveLiquidity` occurs.
+        An example of such is when liquidity is removed, the transfer event of the token removed comes after the
+        `RemoveLiquidity` event.
+        """  # noqa: E501
+        # only run this post decoding rule for transactions interacting the contract.
+        if transaction.to_address not in CURVE_DEPOSIT_CONTRACTS:
```

Comment:
As said before -- this doesn't sound good to me because if a transaction was made via a smart contract wallet, or via some aggregator, or there is basically any intermediate address between the caller and the curve deposit contract then these post decoding rules won't run.

But leaving this question open for @LefterisJP

Reply:
Yes this won't work. We can't check only for direct contract interaction. You also asked me in discord and I said this is not fine.

Also keep in mind the refactoring of post-decoding rules that @yabirgb is going to do in develop so that post decoding only runs for a transaction if the transaction has decoded an event for the given protocol